generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -62,15 +62,20 @@ private boolean mustBeNull(FieldSpec fieldSpec) {
     }
 
     private List<FieldValueSource> getSetRestrictionSources(FieldSpec fieldSpec) {
-        Stream<Object> whitelist = fieldSpec.getSetRestrictions().getWhitelist().stream();
+        List<Object> whitelist = fieldSpec.getSetRestrictions().getWhitelist().stream().collect(Collectors.toList());
 
         if (mayBeNull(fieldSpec)) {
+            if (whitelist.isEmpty()){
+                return Arrays.asList(nullOnlySource);
```

Comment:
Feels wierd that we say sets cannot contain null, but we allow for null to be emitted from them via this method.

Could be clearer if this method emitted an empty collection here, and at line 74.

The calling method `getFieldValueSources` can then append the `nullValueSource` as it sees fit.

i.e. 

_in StandardFieldValueSourceEvaluator_
```
if (fieldSpec.getSetRestrictions() != null && fieldSpec.getSetRestrictions().getWhitelist() != null) {
    List<FieldValueSource> sources = getSetRestrictionSources(fieldSpec);

    if (mayBeNull(fieldSpec)) {
       sources.add(nullOnlySource);
    }

    return sources;
}
```